### ISSUE TITLE
No error for setter with self-referential, implicit any argument

### ISSUE THREAD

**Issue by gabritto** [microsoft/TypeScript#48432]
# Bug Report

A setter can have an argument with a self-referential annotation, and this argument will implicitly have type `any`, with TS producing no errors for it.

### 🔎 Search Terms

setter, implicit any

### 🕗 Version & Regression Information

As far back as 4.3 (when we first let setters/getters in type literals).

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?jsx=0#code/CYUwxgNghgTiAEA3W8CeAueBvAUPf8AziAC7wAOMA9uQEwAUsA5piauSFQGZoB0lNWgEoA3DgC+YoA)

### 💻 Code

```ts
declare var y: {
    set prop2(arg: typeof y.prop2);
};
```

### 🙁 Actual behavior
We don't get any errors, even with `noImplicitAny` set.
But `arg` has implicitly type `any`, because its type annotation refers to the type of the setter itself, and there are no getters.

### 🙂 Expected behavior

An error on `prop2` or `arg`, at least under `noImplicitAny`.

**Comment by ahejlsberg** [comment]
Actually, an error _is_ reported, but the error message is misleading and is reported as a global error because the logic attempts to report on the get accessor, but there isn't one. So, several things are wrong here.
